crates/quickjs-wasm-sys/build.rs

```diff
@@ -1,21 +1,77 @@
-use std::env;
+use anyhow::{anyhow, Result};
 use std::path::PathBuf;
+use std::{env, fs, process};
 
 use walkdir::WalkDir;
 
-fn main() {
-    let this_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
-    let wasi_sdk_path =
-        env::var("QUICKJS_WASM_SYS_WASI_SDK_PATH").unwrap_or(format!("{this_dir}/wasi-sdk"));
-    if !std::path::Path::exists(std::path::Path::new(&wasi_sdk_path)) {
-        panic!(
+const WASI_SDK_VERSION_MAJOR: usize = 20;
+const WASI_SDK_VERSION_MINOR: usize = 0;
+
+fn download_wasi_sdk() -> Result<PathBuf> {
+    let mut wasi_sdk_dir: PathBuf = env::var("CARGO_MANIFEST_DIR")?.into();
+    wasi_sdk_dir.push("wasi-sdk");
+
+    fs::create_dir_all(&wasi_sdk_dir)?;
+
+    let mut archive_path = wasi_sdk_dir.clone();
+    archive_path.push("wasi-sdk.tar.gz");
+
+    // Download archive if necessary
+    if !archive_path.try_exists()? {
+        let file_suffix = match (env::consts::OS, env::consts::ARCH) {
+            ("linux", "x86") | ("linux", "x86_64") => "linux",
+            ("macos", "x86") | ("macos", "x86_64") | ("macos", "aarch64") => "macos",
+            ("windows", "x86") => "mingw-x86",
+            ("windows", "x86_64") => "mingw",
+            other => return Err(anyhow!("Unsupported platform tuple {:?}", other)),
+        };
+
+        let mut archive = fs::File::create(&archive_path)?;
+        reqwest::blocking::get(format!("https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-{}/wasi-sdk-{}.{}-{}.tar.gz", WASI_SDK_VERSION_MAJOR, WASI_SDK_VERSION_MAJOR, WASI_SDK_VERSION_MINOR, file_suffix))?.copy_to(&mut archive)?;
```

Comment:
Nit: I _think_ you can put the variables inside the `{}`s which would make it a little easier to read. See https://doc.rust-lang.org/std/fmt/#usage.

Please do test that this compiles since I haven't tried it on my machine yet:
```suggestion
        reqwest::blocking::get(format!("https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-{WASI_SDK_VERSION_MAJOR}/wasi-sdk-{WASI_SDK_VERSION_MAJOR}.{WASI_SDK_VERSION_MINOR}-{file_suffix}.tar.gz"))?.copy_to(&mut archive)?;
```